### PR TITLE
New case:RHEL-177340 - Snapshot to a non-exist target

### PR DESCRIPTION
**Pull request by aliang123**
Signed-off-by: Aihua Liang <aliang@redhat.com>
id:1979831

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_snapshot_nonexist_target.q35: PASS (35.05 s)

**Comment by aliang123** [comment]
@zhencliu, @ngu-niny, please help to review, thanks.

File: qemu/tests/blockdev_snapshot_nonexist_target.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        if not self.main_vm.is_alive():
+            self.main_vm.create()
+        self.main_vm.verify_alive()
+        try:
+            self.create_snapshot()
+        except QMPCmdError as e:
+            qmp_error_msg = self.params.get("qmp_error_msg")
+            if not re.search(qmp_error_msg, str(e.data)):
+                self.test.fail(str(e))
+
```
Comment (zhencliu) [inline qemu/tests/blockdev_snapshot_nonexist_target.py:19]: we may also need to add 'else' after 'except', in case create_snapshot succeeds unexpectedly

Comment (zhencliu) [reply qemu/tests/blockdev_snapshot_nonexist_target.py:19]: And I guess that create_snapshot may succeed in this case, for the snapshot file has been created successfully. 
we may need to re-define prepare_snapshot_file as:
def prepare_snapshot_file():
pass

Comment (aliang123) [reply qemu/tests/blockdev_snapshot_nonexist_target.py:21]: Have added 'else', please help to check it, zhenchao.

And we don't run prepare_snapshot_file() here, I re-define def run_test(self) here, which just start vm and do snapshot to sn1 (not created).

Comment (zhencliu) [reply qemu/tests/blockdev_snapshot_nonexist_target.py:21]: LGTM

**Comment by zhencliu** [comment]
LGTM

**Comment by ngu-niny** [comment]
(2/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_snapshot_nonexist_target:  PASS (34.78 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**